libs/ledger-live-common/src/families/icon/js-signOperation.ts

```diff
@@ -0,0 +1,124 @@
+import { BigNumber } from "bignumber.js";
+import { Observable } from "rxjs";
+
+import type { IconAccount, Transaction } from "./types";
+import type { Account, Operation, SignOperationEvent } from "@ledgerhq/types-live";
+
+import { withDevice } from "../../hw/deviceAccess";
+import { encodeOperationId } from "../../operation";
+import Icon from "@ledgerhq/hw-app-icon";
+
+import { buildTransaction } from "./js-buildTransaction";
+import { calculateAmount, getNonce } from "./logic";
+import { FeeNotLoaded } from "@ledgerhq/errors";
+import IconService from "icon-sdk-js";
+const { IconUtil, IconConverter } = IconService;
+
+const buildOptimisticOperation = (
+  account: Account,
+  transaction: Transaction,
+  fee: BigNumber,
+): Operation => {
+  const type = "OUT";
+  const value = new BigNumber(transaction.amount).plus(fee);
+  const operation: Operation = {
+    id: encodeOperationId(account.id, "", type),
+    hash: "",
+    type,
+    value,
+    fee,
+    blockHash: null,
+    blockHeight: null,
+    senders: [account.freshAddress],
+    recipients: [transaction.recipient].filter(Boolean),
+    accountId: account.id,
+    transactionSequenceNumber: getNonce(account as IconAccount),
+    date: new Date(),
+    extra: {},
+  };
+
+  return operation;
+};
+
+/**
+ * Adds signature to unsigned transaction. Will likely be a call to Icon SDK
+ */
+const addSignature = (rawTransaction: any, signature: any) => {
```

Comment:
Can you add types here please ? :pray:

Reply:
@hedi-edelbloute I updated the typing in this line and other's comments. thank you